test/api-request-panel.test.js

```diff
@@ -75,6 +79,41 @@ describe('<api-request-panel>', function() {
       editor.execute();
       assert.isTrue(spy.called);
     });
+
+    it('should render server selector', async () => {
+      const element = await basicFixture();
+      assert.exists(element.shadowRoot.querySelector('api-server-selector'));
+    });
+
+    it('should not render server selector with baseUri', async () => {
+      const element = await baseUriFixture();
+      assert.notExists(element.shadowRoot.querySelector('api-server-selector'));
+    });
+  });
+
+  describe('Server selection', () => {
+    let element;
+
+    it('should update selectedServer on api-server-changed event', async () => {
+      element = await basicFixture();
+      element.dispatchEvent(new CustomEvent('api-server-changed', { detail: { value: 'https://www.google.com' } }))
+      assert.equal(element.selectedServer, 'https://www.google.com');
+    });
+
+    describe('with baseUri', () => {
+      beforeEach(async () => {
+        element = await baseUriFixture();
+      });
+
+      it('selectedServer should return baseUri if available', () => {
+        assert.equal(element.selectedServer, 'https://www.example.org');
+      });
+
+      it('baseUri overrides selectedServer set', () => {
+        element.selectedServer = 'https://www.google.com';
+        assert.equal(element.selectedServer, 'https://www.example.org');
+      });
+    })
```

Comment:
This missing a test of what happens when you select a "custom" option in the selector and provide an URI.

Reply:
`api-request-panel` shouldn't care what option is selected, it should need to receive the base URI and apply it to its children. This removes the need to keep track of _which_ server is selected, and only focus on telling its children what the value is

Reply:
But I updated my tests to show this